internal/api/memory.go

```diff
@@ -75,6 +75,13 @@ func copyAndDestroyUnmanagedVector(v C.UnmanagedVector) []byte {
 	return out
 }
 
+func optionalU64ToPtr(val C.OptionalU64) *uint64 {
+	if val.is_some {
+		return (*uint64)(&val.value)
```

Comment:
Do you know where this pointer is pointing to? To the local `val`? To `report` in the caller code?

Would be good to ensure we make a copy somewhere to not risk mutation or free of the underlying value

Reply:
I don't know where it points to, but in Go the GC makes sure that anything you still point to is not free'd. The underlying value is not at risk of mutation, since Go uses pass by value, so the `val` is a copy of the original value. Here's an example: https://play.golang.com/p/CZbASMBFZbV